src/Playwright.Tests/Assertions/PageAssertionsTests.cs

```diff
@@ -0,0 +1,88 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+using static Microsoft.Playwright.Assertions;
+
+namespace Microsoft.Playwright.Tests.Assertions
+{
+    public class PageAssertionsTests : PageTestEx
+    {
+        [PlaywrightTest("playwright-test/playwright.expect.misc.spec.ts", "should support toHaveTitle")]
+        public async Task ShouldSupportToHaveTitleAsync()
+        {
+            await Page.SetContentAsync("<title>  Hello     world</title>");
+            await Expect(Page).ToHaveTitleAsync("Hello  world");
+
+            await Page.SetContentAsync("<title>Bye</title>");
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(() => Expect(Page).ToHaveTitleAsync("Hello", new() { Timeout = 100 }));
+            StringAssert.Contains("Page title expected to be 'Hello'", exception.Message);
+            StringAssert.Contains("But was: 'Bye'", exception.Message);
+            StringAssert.Contains("PageAssertions.ToHaveTitleAsync with timeout 100ms", exception.Message);
+
+            await Page.SetContentAsync("<title>Foo Bar Kek</title>");
+            await Expect(Page).ToHaveTitleAsync(new Regex("^Foo .* Kek$"));
+            await PlaywrightAssert.ThrowsAsync<PlaywrightException>(() => Expect(Page).ToHaveTitleAsync(new Regex("nooo"), new() { Timeout = 100 }));
+
+            await Page.SetContentAsync("<title>Bye</title>");
+            await Expect(Page).Not.ToHaveTitleAsync("Hello");
+        }
+
+        [PlaywrightTest("playwright-test/playwright.expect.misc.spec.ts", "should support toHaveURL")]
+        public async Task ShouldSupportToHaveURLAsync()
+        {
+            await Page.GotoAsync("data:text/html,<div>A</div>");
+            await Expect(Page).ToHaveURLAsync("data:text/html,<div>A</div>");
+
+            await Page.GotoAsync("data:text/html,<div>B</div>");
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(() => Expect(Page).ToHaveURLAsync("wrong", new() { Timeout = 100 }));
```

Comment:
Do we now have both PlaywrightAssert and Playwright.Assert?  If so let's merge former into the latter.

Reply:
PlaywrightAssert is only for us internally, its located inside the Playwright.Tests package: https://cs.github.com/microsoft/playwright-dotnet/blob/82710bccaa67eb4a0bd1035ee332c676398e015d/src/Playwright.Tests/BaseTests/PlaywrightAssert.cs#L32